### ISSUE TITLE
Change default evaluation metric for binary:logitraw to logloss, to be consistent with binary:logistic

### ISSUE THREAD

**Issue by hcho3** [dmlc/xgboost#6618]
https://discuss.xgboost.ai/t/why-is-it-so-slow-with-objective-binary-logitraw/2077

> I have observed a big running time difference with objective=“binary:logistic” and objective=“binary:logitraw” in R with xgboost

**Comment by hcho3** [comment]
Reproducer:
```R
set.seed(1990)
x <- matrix(rnorm(100*2),100,2)
g2 <- sample(c(0,1), 100, replace=TRUE)
library("xgboost")

system.time(fit2 <- xgboost(x, g2, nrounds=500, objective="binary:logitraw", verbose=0))
system.time(fit1 <- xgboost(x, g2, nrounds=500, objective="binary:logistic", verbose=0))
```

**Comment by hcho3** [comment]
Timing information (shown with `verbosity=3`)

`binary:logitraw`:
```
   user  system elapsed
 73.253   0.155   6.498
[20:16:51] ======== Monitor: Learner ========
[20:16:51] Configure: 0s, 1 calls @ 0us

[20:16:51] EvalOneIter: 5.58189s, 500 calls @ 5581893us

[20:16:51] GetGradient: 0.731476s, 500 calls @ 731476us

[20:16:51] PredictRaw: 0.00089s, 500 calls @ 890us

[20:16:51] UpdateOneIter: 0.802032s, 500 calls @ 802032us

[20:16:51] ======== Monitor: GBTree ========
[20:16:51] BoostNewTrees: 0.067723s, 500 calls @ 67723us

[20:16:51] CommitModel: 0.000122s, 500 calls @ 122us

[20:16:51] ======== Monitor: TreePruner ========
[20:16:51] PrunerUpdate: 0.029503s, 500 calls @ 29503us
```

`binary:logitistic`:
```
   user  system elapsed 
  2.523   0.020   0.225 
[20:21:04] ======== Monitor: Learner ========
[20:21:04] Configure: 0s, 1 calls @ 0us

[20:21:04] EvalOneIter: 0.010963s, 500 calls @ 10963us

[20:21:04] GetGradient: 0.002302s, 500 calls @ 2302us

[20:21:04] PredictRaw: 0.000495s, 500 calls @ 495us

[20:21:04] UpdateOneIter: 0.061984s, 500 calls @ 61984us

[20:21:04] ======== Monitor: GBTree ========
[20:21:04] BoostNewTrees: 0.05788s, 500 calls @ 57880us

[20:21:04] CommitModel: 0.000102s, 500 calls @ 102us

[20:21:04] ======== Monitor: TreePruner ========
[20:21:04] PrunerUpdate: 0.024567s, 500 calls @ 24567us
```

**Comment by hcho3** [comment]
The bug also affects the Python package too:
```python
import time
import numpy as np
import xgboost as xgb

rng = np.random.default_rng(seed=1990)
X = rng.normal(loc=0.0, scale=1.0, size=(100, 2))
y = rng.integers(low=0, high=2, size=100)

dtrain = xgb.DMatrix(X, label=y)

params = {
    'objective': 'binary:logitraw',
    'verbosity': 3
}

tstart = time.perf_counter()
bst = xgb.train(params, dtrain, num_boost_round=500, evals=[(dtrain, 'train')])
tend = time.perf_counter()

print(f'Time elapsed: {tend - tstart} sec')

params = {
    'objective': 'binary:logistic',
    'verbosity': 3
}

tstart = time.perf_counter()
bst = xgb.train(params, dtrain, num_boost_round=500, evals=[(dtrain, 'train')])
tend = time.perf_counter()

print(f'Time elapsed: {tend - tstart} sec')
```

`binary:logitraw`:
```
[21:20:20] ======== Monitor: Learner ========                                                                   
[21:20:20] Configure: 0s, 1 calls @ 0us                                                                         
                                                                                                                
[21:20:20] EvalOneIter: 5.70729s, 500 calls @ 5707288us                                                         
                                                                                                                
[21:20:20] GetGradient: 1.23442s, 500 calls @ 1234418us                                                         
                                                                                                                
[21:20:20] PredictRaw: 0.000725s, 500 calls @ 725us                                                             
                                                                                                                
[21:20:20] UpdateOneIter: 1.36944s, 500 calls @ 1369440us                                                       
                                                                                                                
[21:20:20] ======== Monitor: GBTree ========                                                                    
[21:20:20] BoostNewTrees: 0.131131s, 500 calls @ 131131us                                                       
                                                                                                                
[21:20:20] CommitModel: 0.000286s, 500 calls @ 286us                                                            

[21:20:20] ======== Monitor: TreePruner ========
[21:20:20] PrunerUpdate: 0.03241s, 500 calls @ 32410us

Time elapsed: 7.191749425022863 sec
```

`binary:logistic`:
```
[21:20:20] ======== Monitor: Learner ========
[21:20:20] Configure: 0.000265s, 1 calls @ 265us

[21:20:20] EvalOneIter: 0.014799s, 500 calls @ 14799us

[21:20:20] GetGradient: 0.007842s, 500 calls @ 7842us

[21:20:20] PredictRaw: 0.000381s, 500 calls @ 381us

[21:20:20] UpdateOneIter: 0.058419s, 500 calls @ 58419us

[21:20:20] ======== Monitor: GBTree ========
[21:20:20] BoostNewTrees: 0.047663s, 500 calls @ 47663us

[21:20:20] CommitModel: 0.000252s, 500 calls @ 252us

[21:20:20] ======== Monitor: TreePruner ========
[21:20:20] PrunerUpdate: 0.005379s, 500 calls @ 5379us

[21:20:20] ======== Monitor: Learner ========
[21:20:20] ======== Monitor:  ========
[21:20:20] ======== Monitor: TreePruner ========
Time elapsed: 0.12276742304675281 sec
```

**Comment by hcho3** [comment]
This line is 1560x slower when `objective` is set to 'binary:logitraw'`:
https://github.com/dmlc/xgboost/blob/80065d571eacd4c22f316e85262e3197e1eca0bb/src/learner.cc#L1071

objective | Eval
-- | --
binary:logitraw | 5.86334
binary:logistic | 0.003757
Speed difference (times) | 1560.64413

**Comment by hcho3** [comment]
The performance difference was entirely due to the choice of evaluation metric. For some reason, `binary:logitraw` uses AUC as the default evaluation metric, whereas `binary:logistic` uses the log loss. Setting `eval_metric` explicitly eliminates the gap.
```python
import time
import numpy as np
import xgboost as xgb

rng = np.random.default_rng(seed=1990)
X = rng.normal(loc=0.0, scale=1.0, size=(100, 2))
y = rng.integers(low=0, high=2, size=100)

dtrain = xgb.DMatrix(X, label=y)

params = {
    'objective': 'binary:logitraw',
    'eval_metric': 'logloss',
    'verbosity': 3
}

tstart = time.perf_counter()
bst = xgb.train(params, dtrain, num_boost_round=500, evals=[(dtrain, 'train')])
tend = time.perf_counter()

print(f'Time elapsed: {tend - tstart} sec')

params = {
    'objective': 'binary:logistic',
    'eval_metric': 'logloss',
    'verbosity': 3
}

tstart = time.perf_counter()
bst = xgb.train(params, dtrain, num_boost_round=500, evals=[(dtrain, 'train')])
tend = time.perf_counter()

print(f'Time elapsed: {tend - tstart} sec')
```

**Comment by hcho3** [comment]
@lorentzenchr @mayer79 Just following up with #6183. It is my opinion that the default evaluation metric for the `binary:logitraw` objective should also be changed to the logloss so that users are not surprised by the perf difference when switching between `binary:logistic` and `binary:logitraw`.

**Comment by mayer79** [comment]
Good point. We even touched the topic here: https://github.com/dmlc/xgboost/pull/6183#issuecomment-702295697
I had consistency in mind, but calculating auc is indeed time consuming, depending on the implementation!

**Comment by lorentzenchr** [comment]
@hcho3 @mayer79 +1 for the proposed change.

I'm short on time right now. Depending on the urgency, I could work on it a bit later. Although the required change is not so large, #6183 touched 18 different files!

**Comment by trivialfis** [comment]
I believe this time it will be easier.

**Comment by lorentzenchr** [comment]
I welcome the change of the default metric very much.

Thinking about it, however, the API seems a bit strange. AFAIU, the two objectives `binary:logitraw` and `binary:logit` both are fitting the exact same thing (producing the same internal model). The only difference is when you call the `predict` function: `expit(predict(binary:logitraw)) = predict(binary:logit)`. Instead, one could have a `predict_raw` and a `predict` method, or an optional argument `predict(..., raw=False)`. But I doubt that this would have a larger impact in practice.

**Comment by hcho3** [comment]
@lorentzenchr There's already `output_margin` parameter in the `predict()` function. At this point, the `binary:logitraw` objective is a carryover from early versions of XGBoost.

**Comment by lorentzenchr** [comment]
> There's already output_margin parameter in the predict() function

@hcho3 Good to know, thanks.

**Comment by lorentzenchr** [comment]
@hcho3 @trivialfis It's cool to work with you and thanks, by the way, for the fantastic xgboost library!
